src/main/java/com/minecolonies/core/colony/managers/TravellingManager.java

```diff
@@ -0,0 +1,213 @@
+    }
+
+    @Override
+    public Optional<BlockPos> getTravellingTargetFor(final int citizenId)
+    {
+        return Optional.ofNullable(travelerDataMap.get(citizenId)).map(TravelerData::getTarget);
+    }
+
+    @Override
+    public void startTravellingTo(final int citizenId, final BlockPos target, final int travelTimeInTicks)
+    {
+        travelerDataMap.put(citizenId, new TravelerData(citizenId, target, travelTimeInTicks));
+        colony.markDirty();
+    }
+
+    @Override
+    public void finishTravellingFor(final int citizenId)
+    {
+        if (travelerDataMap.containsKey(citizenId))
+        {
+            travelerDataMap.remove(citizenId);
+            colony.markDirty();
+        }
+    }
+
+    @Override
+    public void recallAllTravellingCitizens()
+    {
+        final List<ICitizenData> travellersToRecall = this.travelerDataMap
+          .keySet()
+          .stream()
+          .map(this.colony.getCitizenManager()::getCivilian)
+          .toList();
+
+        this.travelerDataMap.clear();
+        colony.markDirty();
+
+        travellersToRecall.forEach(citizenData -> {
+            final BlockPos spawnHutPos;
+            if (citizenData.getWorkBuilding() != null) {
+                spawnHutPos = citizenData.getWorkBuilding().getPosition();
+            }
+            else
+            {
+                spawnHutPos = colony.getBuildingManager().getTownHall().getPosition();
+            }
+
+            Optional<AbstractEntityCitizen> optionalEntityCitizen = citizenData.getEntity();
+            if (optionalEntityCitizen.isEmpty())
+            {
+                Log.getLogger().warn(String.format("The traveller %d from colony #%d has returned very confused!", citizenData.getId(), colony.getID()));
+                citizenData.setNextRespawnPosition(EntityUtils.getSpawnPoint(colony.getWorld(), spawnHutPos));
+                citizenData.updateEntityIfNecessary();
+                optionalEntityCitizen = citizenData.getEntity();
+            }
+
+            optionalEntityCitizen.ifPresent(abstractEntityCitizen -> TeleportHelper.teleportCitizen(abstractEntityCitizen, colony.getWorld(), spawnHutPos));
+        });
+    }
+
+    public boolean onTick()
+    {
+        travelerDataMap.values().forEach(TravelerData::onTick);
+        return true;
+    }
+
+    @Override
+    public CompoundTag serializeNBT()
+    {
+        final CompoundTag data = new CompoundTag();
+        final ListTag travelerData = new ListTag();
+
+        travelerDataMap.values().stream().map(TravelerData::serializeNBT).forEach(travelerData::add);
+
+        data.put(NbtTagConstants.TRAVELER_DATA, travelerData);
+
+        return data;
+    }
+
+    @Override
+    public void deserializeNBT(final CompoundTag nbt)
+    {
+        final ListTag travelerData = nbt.getList(NbtTagConstants.TRAVELER_DATA, Tag.TAG_COMPOUND);
+        travelerDataMap.clear();
+
+        travelerData
```

Comment:
We had gotten rid of most streams in serialize/deserialize logic as they are nasty to debug when things go wrong and meh on log output too.

Reply:
still

Reply:
Addressed.